lib/filter/tests/test_filters_fop_cmp.c

```diff
@@ -357,6 +357,19 @@ Test(filter, test_type_aware_comparison_objects_are_compared_as_strings)
   cr_assert(evaluate("'$jsonvalue' == list('{\"foo\":\"foovalue\"}')"));
 }
 
+Test(filter, test_type_and_value_comparison_checks_whether_type_and_value_match_completely)
+{
+  cfg_set_version_without_validation(configuration, VERSION_VALUE_4_0);
+  cr_assert(evaluate("$strvalue === $strvalue"));
+  cr_assert(evaluate("$strvalue === string"));
+  cr_assert(evaluate("string(64) === string(64)"));
+  cr_assert_not(evaluate("string(64) !== string(64)"));
+  cr_assert_not(evaluate("string(64) === int64(64)"));
+  cr_assert(evaluate("string(64) !== int64(64)"));
+  cr_assert(evaluate("int32(64) !== int64(64)"));
+  cr_assert_not(evaluate("int32(64) === int64(64)"));
```

Comment:
[optional]

The essential parts are here, but I would add some negative cases too, where the types are matching, but the values are not equal:
```
  cr_assert_not(evaluate("foo === bar"));
  cr_assert_not(evaluate("int64(123) === int64(256)"));
  cr_assert_not(evaluate("123 === 456"));
 }
```

Reply:
I have added these, thanks.